hydra-cluster/bench/Bench/EndToEnd.hs

```diff
@@ -217,25 +221,39 @@ movingAverage confirmations =
               )
    in map average fiveSeconds
 
-createUTxOToCommit :: [Dataset] -> FilePath -> IO [UTxO]
-createUTxOToCommit dataset nodeSocket =
-  forM dataset $ \Dataset{fundingTransaction} -> do
-    submit defaultNetworkId nodeSocket fundingTransaction
-    UTxO.min <$> waitForTransaction defaultNetworkId nodeSocket fundingTransaction
+-- | Distribute 100 ADA fuel and starting funds from faucet for each client in
+-- the dataset.
+seedNetwork :: RunningNode -> Dataset -> IO ()
+seedNetwork node@(RunningNode _ nodeSocket) Dataset{fundingTransaction, clientDatasets} = do
+  -- Submit the funding transaction first
+  putTextLn $ "Funding transaction: " <> decodeUtf8 (prettyPrintJSON fundingTransaction)
+  submit defaultNetworkId nodeSocket fundingTransaction
```

Comment:
This is currently failing, but I don't know why. This is the output received:

Given this genesis UTXO available to the faucet (queried):
```json
{
    "8c78893911a35d7c52104c98e8497a14d7295b4d9bf7811fc1d4e9f449884284#0": {
        "address": "addr_test1vztc80na8320zymhjekl40yjsnxkcvhu58x59mc2fuwvgkc332vxv",
        "value": {
            "lovelace": 900000000000
        }
    }
}
```

and this funding transaction:
```json
{
    "witnesses": {
        "keys": [
            "8200825820ce13cd433cdcb3dfb00c04e216956aeb622dcd7f282b03304d9fc9de804723b258404fde1c834a253d733d3ede9d44023b3843861f361a713ef0ab13f4954430c3494bdcd033a37f602c220afb49dd5dcac29f78dffc8fe5ffd2cb77a0bf3d878107"
        ]
    },
    "isValid": true,
    "body": {
        "outputs": [
            {
                "address": "addr_test1vqgkrhpskez9hu2ydq284vqqluj8mn7hcfncnkhc03q7mssd8e9ty",
                "value": {
                    "lovelace": 202756126265
                }
            },
            {
                "address": "addr_test1vzueffrn5w9huffwvp54wy7hznvp2qnvjk6pln28jfwjwjgqcqexr",
                "value": {
                    "lovelace": 149846552146
                }
            },
            {
                "address": "addr_test1vqux06uxs08q9rhdhvsvng7w5qqvfedjfx2j4cn8nsv4mysl04479",
                "value": {
                    "lovelace": 106934701291
                }
            },
            {
                "address": "addr_test1vztc80na8320zymhjekl40yjsnxkcvhu58x59mc2fuwvgkc332vxv",
                "value": {
                    "lovelace": 440462443269
                }
            }
        ],
        "inputs": [
            "8c78893911a35d7c52104c98e8497a14d7295b4d9bf7811fc1d4e9f449884284#0"
        ],
        "fees": 177029
    },
    "id": "86614a88490f6f815e74ee0a9594f9d62298729f5c76ac74d02a1c12e9a1036d"
}
```

I get this ledger error from the node:
```
       uncaught exception: CardanoClientException
       SubmitException {reason = "TxValidationErrorInMode (ShelleyTxValidationError ShelleyBasedEraAlonzo (ApplyTxError [UtxowFailure (WrappedShelleyEraFailure (UtxoFailure (ValueNotConservedUTxO (Value 0 (fromList [])) (Value 900000000000 (fromList []))))),UtxowFailure (WrappedShelleyEraFailure (UtxoFailure (BadInputsUTxO (fromList [TxInCompact (TxId {_unTxId = SafeHash \"8c78893911a35d7c52104c98e8497a14d7295b4d9bf7811fc1d4e9f449884284\"}) 0]))))])) AlonzoEraInCardanoMode", tx = ShelleyTx ShelleyBasedEraAlonzo (ValidatedTx {body = TxBodyConstr TxBodyRaw {_inputs = fromList [TxInCompact (TxId {_unTxId = SafeHash "8c78893911a35d7c52104c98e8497a14d7295b4d9bf7811fc1d4e9f449884284"}) 0], _collateral = fromList [], _outputs = StrictSeq {fromStrict = fromList [(Addr Testnet (KeyHashObj (KeyHash "9783be7d3c54f11377966dfabc9284cd6c32fca1cd42ef0a4f1cc45b")) StakeRefNull,Value 899899832783 (fromList []),SNothing),(Addr Testnet (KeyHashObj (KeyHash "1161dc30b6445bf14468147ab000ff247dcfd7c26789daf87c41edc2")) StakeRefNull,Value 100000000 (fromList []),SJust (SafeHash "a654fb60d21c1fed48db2c320aa6df9737ec0204c0ba53b9b94a09fb40e757f3"))]}, _certs = StrictSeq {fromStrict = fromList []}, _wdrls = Wdrl {unWdrl = fromList []}, _txfee = Coin 167217, _vldt = ValidityInterval {invalidBefore = SNothing, invalidHereafter = SNothing}, _update = SNothing, _reqSignerHashes = fromList [], _mint = Value 0 (fromList []), _scriptIntegrityHash = SNothing, _adHash = SNothing, _txnetworkid = SNothing}, wits = TxWitnessRaw {_txwitsVKey = fromList [WitVKey' {wvkKey' = VKey (VerKeyEd25519DSIGN "ce13cd433cdcb3dfb00c04e216956aeb622dcd7f282b03304d9fc9de804723b2"), wvkSig' = SignedDSIGN (SigEd25519DSIGN "f22a7e1d3842f822ce54f006f541a0f324c34b484f6b02a47308e13719a398281297c487de998c2b31db447f02cd4e248f9776d8ea44c08acd01af47556bab09"), wvkKeyHash = KeyHash "9783be7d3c54f11377966dfabc9284cd6c32fca1cd42ef0a4f1cc45b", wvkBytes = "\130X \206\DC3\205C<\220\179\223\176\f\EOT\226\SYN\149j\235b-\205\DEL(+\ETX0M\159\201\222\128G#\178X@\242*~\GS8B\248\"\206T\240\ACK\245A\160\243$\195KHOk\STX\164s\b\225\&7\EM\163\152(\DC2\151\196\135\222\153\140+1\219D\DEL\STX\205N$\143\151v\216\234D\192\138\205\SOH\175GUk\171\t"}], _txwitsBoot = fromList [], _txscripts = fromList [], _txdats = TxDatsRaw (fromList []), _txrdmrs = RedeemersRaw (fromList [])}, isValid = IsValid True, auxiliaryData = SNothing})}
```

Reply:
Was missing a `waitForTransaction`